src/vs/workbench/contrib/tasks/browser/abstractTaskService.ts

```diff
@@ -1473,8 +1482,25 @@ export abstract class AbstractTaskService extends Disposable implements ITaskSer
 
 		const execTask = async (task: Task, resolver: ITaskResolver): Promise<ITaskSummary> => {
 			return ProblemMatcherRegistry.onReady().then(() => {
+				// {{ SQL CARBON EDIT }}
+				const taskNodeId = UUID.generateUuid();
+				let taskInfo: TaskInfo = {
```

Comment:
the task info definition is very specific to database, can we make it more generic?

Reply:
This is how SQL Tasks are defined. If we want to generalize that, we would need to change the whole SQL Tasks API. This is just showing SQL Tasks under the generic Tasks View, so database is kind of expected for a SQL Task